lib/archethic.ex

```diff
@@ -116,12 +121,77 @@ defmodule Archethic do
     message = %StartMining{
       transaction: tx,
       welcome_node_public_key: get_welcome_node_public_key(tx_type, welcome_node_key),
-      validation_node_public_keys: Enum.map(validation_nodes, & &1.last_public_key)
+      validation_node_public_keys: Enum.map(validation_nodes, & &1.last_public_key),
+      network_chains_view_hash: NetworkView.get_chains_hash(),
+      p2p_view_hash: NetworkView.get_p2p_hash()
     }
 
-    P2P.broadcast_message(validation_nodes, message)
+    Task.Supervisor.async_stream_nolink(
+      Archethic.TaskSupervisor,
+      validation_nodes,
+      &P2P.send_message(&1, message),
+      ordered: false,
+      on_timeout: :kill_task,
+      timeout: Message.get_timeout(message) + 2000
+    )
+    |> Stream.filter(&match?({:ok, _}, &1))
+    |> Stream.map(fn {:ok, res} -> res end)
+    |> aggregate_start_mining_responses()
+    |> then(fn aggregated_responses ->
+      maybe_start_resync(aggregated_responses)
+
+      if should_forward_transaction?(aggregated_responses, length(validation_nodes)) do
+        forward_transaction(tx, welcome_node_key)
+      else
+        :ok
+      end
+    end)
+  end
+
+  defp aggregate_start_mining_responses(responses) do
```

Comment:
This is better in terms of readability, but if we would like to improve it (for me), something like this would be even better:
```elixir
|> Stream.map(fn {:ok, res} -> res end)
|> Enum.reduce(%{
    ok: 0,
    network_chains_resync_needed: false,
    p2p_resync_needed: false
  }, reduce_start_mining_responses/2)
|> then(...)

defp reduce_start_mining_responses({:ok, %Ok{}}, acc) do
  %{acc | ok: acc.ok + 1}
end

defp reduce_start_mining_responses({:ok, %Error{reason: :network_chains_sync}}, acc) do
   %{acc | network_chains_resync_needed: true}
en

defp reduce_start_mining_responses({:ok, %Error{reason: :p2p_sync}}, acc) do
  %{acc | p2p_resync_needed: true}
end

defp reduce_start_mining_responses({:ok, %Error{reason: :both_sync}}, acc) do
%{ acc | network_chains_resync_needed: true, p2p_resync_needed: true }
end

defp reduce_start_mining_responses(_, acc), do: acc
```